entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -14,6 +14,350 @@ This page does not duplicate the [plan for EF Core 5.0](xref:core/what-is-new/ef
+}
+```
+
+Documentation is tracked by issue [#1979](https://github.com/dotnet/EntityFramework.Docs/issues/1979).
+
+### Migrations: Rebuild SQLite tables
+
+Compared to other database, SQLite is relatively limited in its schema manipulation capabilities. For example, dropping a column from an existing table requires that the entire table be dropped and re-created. EF Core 5.0 Migrations now supports automatic rebuilding the table for schema changes that require it.
+
+For example, imagine we have a `Unicorns` table created for a `Unicorn` entity type:
+
+```c#
+public class Unicorn
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public int Age { get; set; }
+}
+```
+
+```sql
+CREATE TABLE "Unicorns" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_Unicorns" PRIMARY KEY AUTOINCREMENT,
+    "Name" TEXT NULL,
+    "Age" INTEGER NOT NULL
+);
+```
+
+We then learn that storing the age of a unicorn is considered very rude, so let's remove that property, add a new migration, and update the database. This update will fail when using EF Core 3.1 because the column cannot be dropped. In EF Core 5.0, Migrations will instead rebuild the table:
+
+```sql
+CREATE TABLE "ef_temp_Unicorns" (
+    "Id" INTEGER NOT NULL CONSTRAINT "PK_Unicorns" PRIMARY KEY AUTOINCREMENT,
+    "Name" TEXT NULL
+);
+
+INSERT INTO "ef_temp_Unicorns" ("Id", "Name")
+SELECT "Id", "Name"
+FROM Unicorns;
+
+PRAGMA foreign_keys = 0;
+
+DROP TABLE "Unicorns";
+
+ALTER TABLE "ef_temp_Unicorns" RENAME TO "Unicorns";
+
+PRAGMA foreign_keys = 1;
+```
+
+Notice that:
+* A temporary table is created with the desired schema for the new table
+* Data is copied from the current table into the temporary table
+* Foreign key enforcement is switched off
+* The current table is dropped
+* The temporary table is renamed to be the new table
+
+Documentation is tracked by issue [#2583](https://github.com/dotnet/EntityFramework.Docs/issues/2583).
+
+### Table-valued functions
+
+This feature was contributed from the community by [@pmiddleton](https://github.com/pmiddleton). Many thanks for the contribution!
+
+EF Core 5.0 includes first-class support for mapping .NET methods to table-valued functions (TVFs). These functions can then be used in LINQ queries where additional composition on the results of the function will also be translated to SQL.
+
+For example, consider this TVF defined in a SQL Server database:
+
+```sql
+create FUNCTION GetReports(@employeeId int)
+RETURNS @reports TABLE
+(
+	Name nvarchar(50) not null,
+	IsDeveloper bit not null
+)
+AS
+begin
+	WITH cteEmployees AS
+	(
+		SELECT id, name, managerId, isDeveloper
+		FROM employees
+		WHERE id = @employeeId
+		UNION ALL
+		SELECT e.id, e.name, e.managerId, e.isDeveloper
+		FROM employees e
+		INNER JOIN cteEmployees cteEmp ON cteEmp.id = e.ManagerId
+	)
+
+	insert into @reports
+	select name, isDeveloper
+	FROM cteEmployees
+	where id != @employeeId
+
+	return
+end
```

Comment:
```suggestion
CREATE FUNCTION GetReports(@employeeId int)
RETURNS @reports TABLE
(
	Name nvarchar(50) NOT NULL,
	IsDeveloper bit NOT NULL
)
AS
BEGIN
	WITH cteEmployees AS
	(
		SELECT Id, Name, ManagerId, IsDeveloper
		FROM Employees
		WHERE Id = @employeeId
		UNION ALL
		SELECT e.Id, e.Name, e.ManagerId, e.IsDeveloper
		FROM Employees e
		INNER JOIN cteEmployees cteEmp ON cteEmp.Id = e.ManagerId
	)

	INSERT INTO @reports
	SELECT Name, IsDeveloper
	FROM cteEmployees
	WHERE Id != @employeeId

	RETURN
END
```

Reply:
I left the casing as Paul had it, but I agree this aligns better with how we usually show SQL.